hazelcast/src/test/java/com/hazelcast/jet/core/JetTestSupport.java

```diff
@@ -303,6 +305,20 @@ public static long assertJobRunningEventually(HazelcastInstance instance, Job jo
         return executionId;
     }
 
+    /**
+     * Asserts that the {@code job} is eventually SUSPENDED and waits until the suspension cause is set.
+     */
+    public static void assertJobSuspendedEventually(Job job) {
+        assertJobStatusEventually(job, SUSPENDED);
+        assertTrueEventually(() -> {
+            try {
+                assertNotNull(job.getSuspensionCause());
+            } catch (IllegalStateException notSuspended) {
+                Assertions.fail("Suspension cause is not set yet", notSuspended);
+            }
+        });
+    }
+
     public static void assertJobStatusEventually(Job job, JobStatus expected, int timeoutSeconds) {
```

Comment:
We should forbid SUSPENDED status parameter in this (public) method and direct the user to use the method above. Otherwise, we risk creating more flaky tests. 

It will require changing all tests which use `assertJobStatusEventually(job, SUSPENDED);` but is more error-prone for the future.

Reply:
I changed all usages of `assertJobStatusEventually(job, SUSPENDED)` that subsequently tries to retrieve the suspension cause. For usages that are not interested in the suspension cause, no change is required. `assertJobSuspendedEventually` is introduced as a stronger form of `assertJobStatusEventually(job, SUSPENDED)`, just like `assertJobRunningEventually` is a stronger form of `assertJobStatusEventually(job, RUNNING)`.